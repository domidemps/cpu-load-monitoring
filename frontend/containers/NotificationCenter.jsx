/** @jsx jsx */
import {css, jsx} from '@emotion/core'
import {useEffect, useState} from 'react'
import React from 'react'
import Snackbar from '@material-ui/core/Snackbar'
import Alert from '@material-ui/lab/Alert'
import {useSelector} from 'react-redux'

import isEmpty from 'lodash/isEmpty'
import last from 'lodash/last'
import {COLORS} from '../styles/material_ui_raw_theme_file'

export default function NotificationCenter() {
  const [open, setOpen] = useState()
  const events = useSelector(state => state.load.events)

  useEffect(() => {
    setOpen(true)
  }, [events])

  if (isEmpty(events)) {
    return null
  }

  const handleClose = (event, reason) => {
    if (reason === 'clickaway') {
      return
    }
    setOpen(false)
  }

  const isHeavyLoad = last(events).type === 'heavy'

  return (
    <Snackbar
      open={open}
      autoHideDuration={10000}
      onClose={handleClose}
      anchorOrigin={{vertical: 'top', horizontal: 'right'}}>
      <Alert
        onClose={handleClose}
        severity={isHeavyLoad ? 'error' : 'success'}
        css={
          isHeavyLoad
            ? css`
                border: solid ${COLORS.heavyLoad};
              `
            : css`
                border: solid ${COLORS.normalLoad};
              `
        }>
        {isHeavyLoad ? 'System under heavy CPU load' : 'System recovered from heavy CPU load'}
      </Alert>
    </Snackbar>
  )
}
