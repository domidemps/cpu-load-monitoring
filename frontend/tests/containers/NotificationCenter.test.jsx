import React from 'react'
import configureStore from 'redux-mock-store'
import thunk from 'redux-thunk'
import {Provider} from 'react-redux'
import renderer from 'react-test-renderer'

import NotificationCenter from '../../containers/NotificationCenter'

describe('NotificationCenter', () => {
  let store

  it('renders correctly if no events', () => {
    store = configureStore([thunk])({
      load: {
        events: [],
      },
    })
    expect(
      renderer
        .create(
          <Provider store={store}>
            <NotificationCenter />
          </Provider>,
        )
        .toJSON(),
    ).toMatchSnapshot()
  })
})
