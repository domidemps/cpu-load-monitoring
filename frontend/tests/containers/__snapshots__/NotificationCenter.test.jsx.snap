// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotificationCenter renders correctly if no events 1`] = `null`;
